### PR TITLE
[Feat] Enable State Dict For Textual Inversion Loader

### PR DESCRIPTION
**Pull request by ghunkins**
**Enable passing a pre-loaded state dict via the `TextualInversionLoaderMixin`.**

Benchmarked the implementation of [`LoraLoaderMixin`](https://github.com/huggingface/diffusers/blob/29b1325a5ae28fa8d7f459b372582287ffc571e5/src/diffusers/loaders.py#L771).

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Hey @ghunkins,

What exactly is the use case to allow for this? Would be great to better understand why we need it :-)

**Comment by ghunkins** [comment]
Tried to re-run the tests to get CI to pass, but I am guessing #3449 is the fix for that.

Requesting review before this gets lost, thanks @patrickvonplaten @patil-suraj @williamberman! 🙏

File: src/diffusers/loaders.py [hunk 1]
```diff
@@ -638,16 +640,38 @@ def load_textual_inversion(
...
 
         for pretrained_model_name_or_path, token in zip(pretrained_model_name_or_paths, tokens):
-            # 1. Load textual inversion file
-            model_file = None
-            # Let's first try to load .safetensors weights
-            if (use_safetensors and weight_name is None) or (
-                weight_name is not None and weight_name.endswith(".safetensors")
-            ):
-                try:
+            if not isinstance(pretrained_model_name_or_path, dict):
```
Comment (patrickvonplaten) [inline src/diffusers/loaders.py:643]: why changing the order here? Can we revert this maybe?

**Comment by patrickvonplaten** [comment]
Hey @ghunkins,

Generally, this PR works for me! Could you maybe also add a quick test to this test function: https://github.com/huggingface/diffusers/blob/6dd3871ae05bd2a7d52c637e14db17887868aee9/tests/pipelines/test_pipelines.py#L494 ?

**Review by sayakpaul — COMMENTED** [review]
Changes look good for me. Like @patrickvonplaten mentioned, would be nice to add some tests to ensure that the changes are robust. 

I'd also like a review from @pdoane since they helped us with batch loading of text inversion parameters (https://github.com/huggingface/diffusers/issues/3193).

**Review by pdoane — APPROVED** [review]
Looks good!

**Comment by patrickvonplaten** [comment]
@ghunkins let me know if you need more clarification on how to add a test :-)

**Comment by ghunkins** [comment]
@patrickvonplaten Great call, tests are added! Let me know if this works.

cc @sayakpaul @pdoane

**Review by sayakpaul — APPROVED** [review]
What a comprehensive coverage of tests! Love'em, just love'em!

**Comment by ghunkins** [comment]
Great, thanks @sayakpaul! So much easier when great tests are already there to build off of and benchmark against 🚀

Let me know if there's anything else I can do to get this across the finish line. Looks like only contributors with `write` access can merge.

cc @patrickvonplaten @pdoane

**Comment by patrickvonplaten** [comment]
Very cool PR, good job! @ghunkins

**Pull request merged**